### PR TITLE
Bump Kubernetes

### PR DESCRIPTION
**Pull request by fabianofranz**

### CODE REVIEW

File: pkg/cmd/cli/cmd/factory.go [hunk 1]
```diff
@@ -5,14 +5,11 @@ import (
```
Comment (fabianofranz) [inline pkg/cmd/cli/cmd/factory.go]: @smarterclayton please take a look at this file - I'm missing something that I can't figure.

**Comment by fabianofranz** [comment]
@smarterclayton Now fixing tests, server starts and the client works. Any clues about the messages below? Can be a coincidence but they are all objects with camelcase names.

```
E0116 01:26:41.789743    2087 reflector.go:80] Failed to list *api.ImageRepository: imageRepositories "" not found
E0116 01:26:41.789899    2087 reflector.go:80] Failed to list *api.BuildConfig: buildConfigs "" not found
E0116 01:26:41.798774    2087 reflector.go:80] Failed to list *api.ImageRepository: imageRepositories "" not found
E0116 01:26:41.815699    2087 reflector.go:80] Failed to list *api.DeploymentConfig: deploymentConfigs "" not found
E0116 01:26:41.821336    2087 reflector.go:80] Failed to list *api.DeploymentConfig: deploymentConfigs "" not found
```

**Comment by smarterclayton** [comment]
If you updated RESTMapper you need to initialize our v1beta1 with case sensitive strings (whatever upstream uses for v1beta1)

**Comment by smarterclayton** [comment]
Or you're using the wrong value of legacy for initializing our RESTClient

**Comment by fabianofranz** [comment]
That's it, config.LegacyBehavior = (config.Version == "v1beta1" || config.Version == "v1beta2")

----- Original Message -----

**Comment by smarterclayton** [comment]
Is GitHub confused by your pull request (seeing dup commits)

**Comment by fabianofranz** [comment]
No, there are really multiple commits with the same name. I'll rebase once all tests are fixed in one bump and one refactoring commit. Notice the two bump commits are different (for two different Kube versions), but we can just keep the latest one.

----- Original Message -----

**Comment by mfojtik** [comment]
@fabianofranz i'm working now to fix some if the failures from test-go.sh ;-) pls ping me on IRC once you get back to this PR ;-)

**Comment by fabianofranz** [comment]
[test]

**Comment by fabianofranz** [comment]
@smarterclayton Good to review. The Travis failure seem to be unrelated, I'll run the tests locally once again to make sure.

**Comment by liggitt** [comment]
@fabianofranz looks like you pulled in an old version of go-bindata from kubernetes, which will break our build. You need to revert  accidentally... need to "revert" github.com/jteeuwen/go-bindata to the newer version we already had

**Comment by smarterclayton** [comment]
There should be an UPSTREAM tagged commit that does this, and also excludes Kube go assets.  Please search the commit log for UPSTREAM and include any that weren't already merged.

**Comment by liggitt** [comment]
@smarterclayton in which repo/branch?

**Comment by liggitt** [comment]
@fabianofranz in case you can't find what clayton was referring to, I opened a PR against your branch that does the go-bindata bump

**Comment by smarterclayton** [comment]
Master, delivered the last time I rebased.  the whole commit is necessary (might even be two).

**Comment by smarterclayton** [comment]
Fabiano, can you be sure to brief mark and Jhon on this part of the process and get it added into the rebase instructions?

**Comment by liggitt** [comment]
All I can find in godeps for that dependency are:
```
2624119a - jessica bumped to v3.0.7-2-gf94581b (correct version)
0de2fe33 - jessica bumped to 3.0.7
a3d5e674 - jessica bumped to 3.0.6
63ec7fa8 - jessica initial add of the framework
```

**Comment by smarterclayton** [comment]
I didn't change godeps, I changed godeps json.

**Comment by liggitt** [comment]
Right... those are the only version changes for go-bindata in godeps.json

**Comment by smarterclayton** [comment]
Hrm, the bi data commit must have gotten squashed.  We should not do that this time (preserve on top of bump, prefix with UPSTREAM:)

**Comment by fabianofranz** [comment]
@smarterclayton There are 3 specifically tagged UPSTREAM:

https://github.com/openshift/origin/commit/7c2920057f18c14e5415a7224ea85b184a293771
https://github.com/openshift/origin/commit/10396b2df1068cf31bc0092307f043cdf02aab05
https://github.com/openshift/origin/commit/73c559e4c6278e1451561ffb8a27e5a316fa061c

The first one was successfully cherry-pick, the other ones were empty (probably on Kube already).

**Comment by smarterclayton** [comment]
We need to fix go-bindata somehow.

**Comment by fabianofranz** [comment]
@smarterclayton I'll brief @jwhonce and @markturansky about that and other details I found in the process, adding a task to the Trello card.

**Comment by fabianofranz** [comment]
[test]

**Comment by fabianofranz** [comment]
@smarterclayton @liggitt all comments addressed.

**Comment by smarterclayton** [comment]
Can you squash michal's changes into one commit for him and one for you (so we know who to blame)

**Comment by fabianofranz** [comment]
@smarterclayton squashed @mfojtik into one. I kept mine as 3 separate commits (one for the bump, one for the refactorings and one UPSTREAM so that we don't loose it in the next rebase).

**Comment by mfojtik** [comment]
LGTM :+1:

**Comment by smarterclayton** [comment]
LGTM [merge]

**Pull request merged**